test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,69 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+template <typename T, typename = std::enable_if<std::is_floating_point_v<T>>>
+struct FPUtils
+{
+    using Limits = std::numeric_limits<T>;
+
+    static constexpr auto NumMantissaBits = Limits::digits - 1;
+    static constexpr auto MantissaMask = (uint64_t{1} << NumMantissaBits) - 1;
+
+    static constexpr auto CanonicalNaNPayload = uint64_t{1} << (NumMantissaBits - 1);
+};
+
+template <typename T, typename = std::enable_if<std::is_floating_point_v<T>>>
+uint64_t as_uint64(T value) noexcept
+{
+    constexpr auto num_bytes = sizeof(T);
+    uint8_t bytes[num_bytes];
+    __builtin_memcpy(bytes, &value, num_bytes);
+
+    uint64_t u = 0;
+    for (size_t i = 0; i < num_bytes; ++i)
+        u |= uint64_t{bytes[i]} << (i * 8);  // little-endian
```

Comment:
Could do `i += 8` each iteration instead of `i * 8`, but maybe not very important

Reply:
Nah.